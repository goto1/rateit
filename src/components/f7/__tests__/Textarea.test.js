import React from "react";
import Textarea from "../Textarea";
import renderer from "react-test-renderer";
import { shallow } from "enzyme";
import "jest-enzyme";

const props = {
  icon: "info",
  name: "about",
  placeholder: "Tell us about yourself...",
  onChange: jest.fn()
};

describe("Textarea component", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<Textarea {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders a textarea element with a specified name attribute", () => {
    const wrapper = shallow(<Textarea {...props} />);
    expect(wrapper.find("textarea")).toHaveProp("name", props.name);
  });

  it("calls a function when value changes", () => {
    const wrapper = shallow(<Textarea {...props} />);
    const { onChange } = props;

    wrapper.find("textarea").simulate("change");
    expect(onChange.mock.calls.length).toBe(1);
  });
});
