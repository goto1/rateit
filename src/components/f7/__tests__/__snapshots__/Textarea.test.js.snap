// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Textarea component renders correctly 1`] = `
<li
  className="align-top"
>
  <div
    className="item-content"
  >
    <div
      className="item-media"
    >
      <i
        className="material-icons icon"
      >
        info
      </i>
    </div>
    <div
      className="item-inner"
    >
      <div
        className="item-input"
      >
        <textarea
          name="about"
          onChange={[Function]}
          placeholder="Tell us about yourself..."
        />
      </div>
    </div>
  </div>
</li>
`;
